src/Explorer/Panes/AddCollectionPanel.tsx

```diff
@@ -546,6 +548,90 @@ export class AddCollectionPanel extends React.Component<AddCollectionPanelProps,
                   }
                 }}
               />
+              {userContext.apiType === "SQL" &&
+                this.state.subPartitionKeys.map((subPartitionKey: string, index: number) => {
+                  return (
+                    <Stack style={{ marginBottom: 8 }} key={`uniqueKey${index}`} horizontal>
+                      <div
+                        style={{
+                          width: "20px",
+                          border: "solid",
+                          borderWidth: "0px 0px 1px 1px",
+                          marginRight: "5px",
+                        }}
+                      ></div>
+                      <input
+                        type="text"
+                        id="addCollection-partitionKeyValue"
+                        key={`addCollection-partitionKeyValue_${index}`}
+                        aria-required
+                        required
+                        size={40}
+                        tabIndex={index > 0 ? 1 : 0}
+                        className="panelTextField"
+                        autoComplete="off"
+                        placeholder={this.getPartitionKeyPlaceHolder(index)}
+                        aria-label={this.getPartitionKeyName()}
+                        pattern={".*"}
+                        title={""}
+                        value={subPartitionKey}
+                        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                          if (!this.state.subPartitionKeys[index] && !event.target.value.startsWith("/")) {
+                            const subPartitionKeys = this.state.subPartitionKeys.map(
```

Comment:
It looks like we don't really need a `map` loop here right? We can directly update `this.state.subPartitionKeys[index]`:

const newSubPartitionKeys = [...this.state.subPartitionKeys];
newSubPartitionKeys[index] = !this.state.subPartitionKeys[index] && !event.target.value.startsWith("/") ? "/" + 		 
    event.target.value.trim() : event.target.value.trim();
this.setState({ subPartitionKeys: newSubPartitionKeys });